src/index.coffee

```diff
@@ -94,8 +115,31 @@ reinstall_all = (options = {}, packages) ->
 
   return if packages then curried packages else curried
 
+shrinkwrap = (packages, options = {}) ->
+  shrinkwrap_json =
+    dependencies: {}
+
+  for pkg in packages
+    [ whole, git_at_github, name, branch] = pkg.match /^(.+?):(.+?)(?:\.git)(?:#(.+))?$/
+    ref = "refs/heads/#{branch}"
```

Comment:
Neither shas nor tags would work with these semantics. Perhaps we could add some error handling for those cases, since shas and tags would not need to be shrinkwrapped, since they always point to the same point in time.

It might make sense to simply not include gitDependencies that were pointed to a sha or tag-- or for the sake of being explicit in the shrinkwrap, to resolve the tag to a sha, and then include shas for every gitDependency